File: Python3/reverse_linked_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        reverse = None
        
        while head:
            next = head.next
            head.next = reverse
            reverse = head
            head = next
            
        return reverse

File: C/reverse_linked_list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* reverseList(struct ListNode* head){
    struct ListNode* reverse = NULL;
        
    while(head != NULL) {
        struct ListNode* remain = head->next;
        head->next = reverse;
        reverse = head;
        head = remain;
    }
    return reverse;
}
